### PR TITLE
Implement networking for OCI runtime

### PR DESCRIPTION
**Pull request by bduffany**
Not much interesting new code is being added here except a new `iptables` rule in `SetupVethPair` to prevent traffic between containers. Everything else is mostly plumbing / tests.

**Related issues**: https://github.com/buildbuddy-io/buildbuddy/pull/7093

### CODE REVIEW

**Review by fmeum — APPROVED** [review]

File: server/util/networking/networking.go [hunk 1]
```diff
@@ -443,23 +429,56 @@ func SetupVethPair(ctx context.Context, netNamespace string) (_ *VethPair, err e
...
+	iptablesRules := [][]string{
+		// Allow forwarding traffic from the host side of the veth pair to the
+		// device associated with the configured route prefix (usually the
+		// default route). This is necessary on hosts with default-deny policies
+		// in place.
+		{"FORWARD", "-i", veth1, "-o", device, "-j", "ACCEPT"},
+
+		// Drop any traffic from the namespace that is targeting another
+		// namespace.
+		{"FORWARD", "-s", network.CIDR(), "-d", containerNetworkingCIDR, "-j", "DROP"},
```
Comment (vadimberezniker) [inline server/util/networking/networking.go:441]: Containers shouldn't be addressing any private addresses so you might as well block all the ranges in `PrivateIPRanges`.

**Review by vadimberezniker — APPROVED** [review]

**Review by iain-macdonald — APPROVED** [review]
LGTM, Approval

File: server/util/networking/networking.go (hunk 1 above)
Comment (bduffany) [reply server/util/networking/networking.go:441]: We also have `executor.blackhole_private_ranges` which should disallow traffic to private ranges, but it felt somewhat incidental that `containerNetworkingCIDR` happens to be one of the private IP ranges so I thought it was worth an explicit rule. 

If we want to consolidate the handling of these DROP rules and maybe get rid of that blackhole_private_ranges flag, I can do that in a separate PR - wdyt?

File: server/testutil/testnetworking/testnetworking.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testfs"
+	"github.com/stretchr/testify/require"
+)
+
+// Setup sets up the test to be able to call networking functions.
+// If skips the test if the required net tools aren't available.
```
[the remark this answers is not in the document]
Comment (bduffany) [reply server/testutil/testnetworking/testnetworking.go]: woops, fixed

File: server/util/networking/networking.go (hunk 1 above)
Comment (bduffany) [reply server/util/networking/networking.go:441]: (Will discuss more offline and follow up in another PR)

**Pull request merged**